napalm/junos/junos.py

```diff
@@ -235,14 +237,75 @@ def compare_config(self):
             return diff.strip()
 
     def commit_config(self, confirmed=None, message=None):
-        """Commit configuration."""
-        if confirmed is not None:
-            raise NotImplementedError
+        """
+        Commit configuration.
+
+        if confirmed is passed, random string is generated for the commit log
+        """
+        pending_commit_string = None
+        if confirmed:
+            pending_commit_string = str(uuid.uuid4()).split('-')[0]
+            if message:
+                message = '{}_napalm_confirm_{}'.format(message, pending_commit_string)
```

Comment:
I don't think it's a good idea to commit with a different message than the user requested.